themes/socialblue/color/preview.js

```diff
@@ -64,10 +64,35 @@
       $colorPreview.find('.body-text a:not(.btn)').css('color', $colorPalette.find('input[name="palette[brand-link]"]').val());
 
       // Hero toggle background.
-      $colorPreview.find('.color-preview-hero .switch').once().click(function() {
-        $('.color-preview-hero .switch .lever').toggleClass("lever-on");
-        $('.color-preview-hero .cover').toggleClass("cover-img")
-          .toggleClass("cover-img-gradient");
+      var localStorage =  window.localStorage;
+      var colorPreviewLever = $('.color-preview-hero .switch .lever');
+      var colorPreviewCover = $('.color-preview-hero .cover');
+
+      if (localStorage.getItem('lever-on') &&
+        localStorage.getItem('set-settings-theme')) {
+          colorPreviewLever.addClass('lever-on');
+          colorPreviewCover.addClass('cover-img')
+            .addClass('cover-img-gradient');
+      }
+
+      $colorPreview.find('.color-preview-hero .switch').once()
+        .click(function() {
+          colorPreviewLever.toggleClass('lever-on');
+          colorPreviewCover.toggleClass('cover-img')
+            .toggleClass('cover-img-gradient');
+
+          if (colorPreviewLever.hasClass('lever-on')) {
+            localStorage.setItem('lever-on', '1');
+          }
+          else {
+            localStorage.removeItem('lever-on');
+          }
+        }
+      );
+
+      var settingThSub = $('.system-theme-settings .form-actions .form-submit');
+      settingThSub.on('click', function () {
+        localStorage.setItem('set-settings-theme', '1');
```

Comment:
I think we can just remove this setting entirely and just persist the user's preference regardless of whether they actually saved. After the first time they've saved this page that will be the behaviour anyway.

Reply:
If I understood you right, you want save it without localStorage? save on color.inc file?

Reply:
Sorry I didn't really word it clearly (had to re-figure it out myself as well). The localStorage is already updated when the user toggles the setting (see https://github.com/goalgorilla/open_social/pull/1356/files/67a698be713c89e7b4677b853a2f630ffa31b8ec#diff-f6039e2aa13ec5f04b7b67d2a88d39a5R84 ). 

There's no real reason to also do this when the user saves the form. So we can remove line 95 :)